### PR TITLE
redux of pr 7398 for ssh-agent key fix

### PR DESCRIPTION
**Pull request by zmt**
This is probably easier...

### CODE REVIEW

File: plugins/commands/ssh_config/command.rb [hunk 1]
```diff
@@ -37,13 +37,20 @@ def execute
...
             proxy_command: ssh_info[:proxy_command],
             ssh_command:   ssh_info[:ssh_command],
             forward_env:   ssh_info[:forward_env],
           }
+          if ssh_info[:private_key_path]
+            variables['private_key_path'] = ssh_info[:private_key_path]
+          end
+          if ssh_info[:log_level]
+            variables['log_level'] = ssh_info[:log_level]
+          end
```
Comment (chrisroberts) [inline plugins/commands/ssh_config/command.rb]: Any reason for conditionally setting these variable entries? If there's not, can they be placed back into the hash construction?

File: plugins/synced_folders/rsync/helper.rb [hunk 2]
```diff
@@ -63,7 +63,11 @@ def self.rsync_single(machine, ssh_info, opts)
         opts[:owner] ||= ssh_info[:username]
         opts[:group] ||= ssh_info[:username]
 
+        # set log level
+        log_level = ssh_info[:log_level] || "FATAL"
+
```
Comment (chrisroberts) [inline plugins/synced_folders/rsync/helper.rb]: This `log_level` looks to be extraneous here, or was logging level meant to be set within the options?

**Comment by chrisroberts** [comment]
@zmt Looks great! Left notes on a couple small things. If you could address those, that would be great. Thanks!

**Comment by chrisroberts** [comment]
Applied some updates to this PR via #7703. Thanks for getting this PR together!

**Comment by raatti** [comment]
This is still not fixed in Released version, when is this getting out?

File: plugins/commands/ssh_config/command.rb (hunk 1 above)
Comment (zmt) [inline plugins/commands/ssh_config/command.rb]: yes - we wanted to do ssh-agent only, which would mean that ssh_info[:private_key_path] would not be set

File: plugins/synced_folders/rsync/helper.rb (hunk 2 above)
Comment (zmt) [inline plugins/synced_folders/rsync/helper.rb]: I don't recall at the moment, but this was either copied from older code or prevented errors from log_level not being set in my testing way back when.

**Comment by zmt** [comment]
@raatti This went out with 1.8.6 on 9/27/16 apparently.

**Pull request closed without merging**